Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp

```diff
@@ -242,14 +290,27 @@ void AuthenticatorCoordinator::create(const Document& document, CredentialCreati
         });
     }
 
-    auto callback = [weakThis = WeakPtr { *this }, promise = WTFMove(promise), abortSignal = WTFMove(abortSignal)] (AuthenticatorResponseData&& data, AuthenticatorAttachment attachment, ExceptionData&& exception) mutable {
+    auto& globalObject = *JSC::jsCast<JSDOMGlobalObject*>(document.globalObject());
+    auto requestJSON = toInspectorJSON(globalObject, *createOptions.publicKey);
+
+    auto ceremonyId = Inspector::IdentifiersFactory::createIdentifier();
+    InspectorInstrumentation::frameDidStartWebAuthenticationOperation(*frame, ceremonyId, requestJSON.releaseNonNull(), initiatorStackTrace.copyRef());
+
+    auto callback = [weakThis = WeakPtr { *this }, promise = WTFMove(promise), abortSignal = WTFMove(abortSignal), requestJSON = WTFMove(requestJSON), weakDocument = WeakPtr { document }, initiatorStackTrace = WTFMove(initiatorStackTrace), ceremonyId] (AuthenticatorResponseData&& data, AuthenticatorAttachment attachment, ExceptionData&& exception) mutable {
         if (abortSignal && abortSignal->aborted()) {
             promise.reject(Exception { ExceptionCode::AbortError, "Aborted by AbortSignal."_s });
             return;
         }
 
         if (auto response = AuthenticatorResponse::tryCreate(WTFMove(data), attachment)) {
-            promise.resolve(PublicKeyCredential::create(response.releaseNonNull()).ptr());
+            auto publicKeyCredential = PublicKeyCredential::create(response.releaseNonNull());
+            RefPtr document = weakDocument.get();
+            if (RefPtr frame = document ? document->frame() : nullptr) {
```

Comment:
is there a reason you're trying to get and pass the `LocalFrame` instead of just the `Document`?  `InspectorInstrumentation` also supports the latter (and if you propagate the `Document` into `InspectorPageAgent` then you can get the `LocalFrame` at that point)